packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -174,7 +187,33 @@ class RefreshIndicator extends StatefulWidget {
     this.semanticsValue,
     this.strokeWidth = RefreshProgressIndicator.defaultStrokeWidth,
     this.triggerMode = RefreshIndicatorTriggerMode.onEdge,
-  }) : _indicatorType = _IndicatorType.adaptive;
+  })  : _indicatorType = _IndicatorType.adaptive,
+        onStatusChange = null;
+
+  /// {@tool dartpad}
+  /// Creates [RefreshIndicator] with no spinner and calls `onRefresh` when
+  /// successfully armed by drag event. Events can be optionally listened
+  /// by using `onStatusChange` callback.
+  ///
+  /// ** See code in examples/api/lib/material/refresh_indicator/refresh_indicator.2.dart **
+  /// {@end-tool}
+  const RefreshIndicator.noSpinner({
+    super.key,
+    required this.child,
+    required this.onRefresh,
+    this.onStatusChange,
+    this.notificationPredicate = defaultScrollNotificationPredicate,
+    this.semanticsLabel,
+    this.semanticsValue,
+    this.triggerMode = RefreshIndicatorTriggerMode.onEdge,
+  })  : _indicatorType = _IndicatorType.noSpinner,
+        // these parameters aren't used for [_IndicatorType.noSpinner],
+        // setting default values
```

Comment:
This comment should probably be more descriptive, i.e

`These parameters aren't used for [_IndicatorType.noSpinner] because... So, their default values are used.`